django/contrib/postgres/fields/array.py

```diff
@@ -16,6 +18,32 @@
 __all__ = ['ArrayField']
 
 
+class ArrayLiteral:
+    """
+    An object which psycopg2 serializes into an array, but is hashable and
+    comparable like a tuple.
+    """
+
+    def __init__(self, value):
+        self.value = tuple(value)
+
+    def __conform__(self, protocol):
+        if protocol is psycopg2.extensions.ISQLQuote:
+            return psycopg2.extensions.adapt(list(self.value))
+        return None
+
+    def __hash__(self):
+        return hash(self.value)
+
+    def __eq__(self, other):
+        if not isinstance(other, ArrayLiteral):
+            return NotImplemented
+        return self.value == other.value
+
+    def __repr__(self):
+        return '<ArrayLiteral {0.value!r}>'.format(self)
```

Comment:
This is untested.

Reply:
It isn't really needed, `Query` doesn't have a `__repr__` for example so I removed it